server/daemon/broker/redis/redis.go

```diff
@@ -0,0 +1,93 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-redis/redis/v8"
+)
+
+const (
+	redisPort  = 6379
+	channelKey = "mqbenchmark:chan"
+)
+
+// PubSubPeer implements the peer interface for Redis Pub/Sub
+// for more info on RedisPubSub see https://redis.io/topics/pubsub
+type PubSubPeer struct {
+	conn     *redis.Client
+	pubsub   *redis.PubSub
+	messages <-chan *redis.Message
+	send     chan []byte
+	errors   chan error
+	done     chan bool
+}
+
+// NewPeer creates a peer used for communicating with Redis
+func NewPeer(host string) (*PubSubPeer, error) {
+	p := PubSubPeer{}
+	p.conn = redis.NewClient(&redis.Options{
+		Addr: fmt.Sprintf("%s:%d", host, redisPort),
+	})
+	p.send = make(chan []byte)
+	p.errors = make(chan error)
+	p.done = make(chan bool)
+	return &p, nil
+}
+
+// Subscribe prepares the peer to consume messages.
+func (p *PubSubPeer) Subscribe() error {
+	p.pubsub = p.conn.Subscribe(context.Background(), channelKey)
```

Comment:
Is this asynchronous by default? Could be the cause of poor results.

Reply:
I'm not sure. I'll have to look into it - though it might be. In terms of perf, wouldn't subscribe only be called once per consumer?